tests/fixtures/database.py

```diff
@@ -616,8 +643,23 @@ def collection(
 
         if library and library not in collection.associated_libraries:
             collection.associated_libraries.append(library)
+
+        if inactive:
+            self.make_collection_inactive(collection)
         return collection
 
+    @staticmethod
+    def make_collection_inactive(collection: Collection):
+        """Make a collection inactive using some settings that will make it so."""
+        collection.integration_configuration.settings_dict.update(
+            {
+                "subscription_activation_date": datetime.date(2000, 12, 31),
+                "subscription_expiration_date": datetime.date(1999, 1, 1),
+            }
+        )
+        flag_modified(collection.integration_configuration, "settings_dict")
+        assert not collection.is_active
```

Comment:
Minor: Might be nice to use the methods on the integration to make this settings change, since it does a little more validation.

```suggestion
    def make_collection_inactive(self, collection: Collection) -> None:
        """Make a collection inactive using some settings that will make it so."""
        protocol_cls = self._goal_registry_mapping[Goals.LICENSE_GOAL][collection.protocol]
        protocol_cls.settings_update(
            collection.integration_configuration,
            {
                "subscription_activation_date": datetime.date(2000, 12, 31),
                "subscription_expiration_date": datetime.date(1999, 1, 1),
            },
            merge=True
        )
        assert not collection.is_active
```